Salwador/UIComponents/GenerateImage/GenerateImageViewModel.swift

```diff
@@ -15,35 +15,54 @@ import UIKit
     @Published var isLoading: Bool = false
     @Published var hasError: Bool = false
     @Published var errorMsg: String = ""
-    @Published var apiKeyFileName: String = ""
+    @Published var isPopListShown: Bool = false
```

Comment:
Idea: i would suggest to consolidate all of this here. 

https://www.hackingwithswift.com/books/ios-swiftui/switching-view-states-with-enums